### PR TITLE
implement CPU contrib OP Attention

### PR DESCRIPTION
**Pull request by fs-eire**

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/bert/attention.cc [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+  // Use GEMM for fully connection.
+  int m = batch_size * sequence_length;
+  int n = 3 * hidden_size;
+  int k = hidden_size;
+
+  AllocatorPtr allocator;
+  ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&allocator));
+
+  concurrency::ThreadPool* tp = context->GetOperatorThreadPool();
```
Comment (pranavsharma) [inline onnxruntime/contrib_ops/cpu/bert/attention.cc]: tp can be null.

Comment (fs-eire) [reply onnxruntime/contrib_ops/cpu/bert/attention.cc]: Thanks for the remind. All call paths are checked.

File: onnxruntime/contrib_ops/cpu/bert/attention.cc [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  // STEP.5: out_tmp(B, N, S, H) = P(B, N, S, S) x V(B, N, S, H)
+  auto out_tmp_data = allocator->Alloc(batch_size * num_heads_ * sequence_length * head_size * element_size);
+  BufferUniquePtr out_tmp_buffer(out_tmp_data, BufferDeleter(allocator));
+  {
+    int offset_p = 0;
+    int offset_p_increment = sequence_length * sequence_length;
+    int offset_V = 0;
+    int offset_V_increment = sequence_length * head_size;
+
+    for (int b_i = 0; b_i < batch_size; b_i++) {
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/bert/attention.cc]: Can we use batch GEMM to improve performance? Like https://software.intel.com/en-us/articles/introducing-batch-gemm-operations

Comment (fs-eire) [reply onnxruntime/contrib_ops/cpu/bert/attention.cc]: I am trying different ways to improve the performance of Attention, which including Gemm and MatMul as well. That will come later (not going to be a part of this PR)

**Pull request merged**